editor/src/java/com/defold/editor/luart/DefoldBaseLib.java

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package com.defold.editor.luart;
+
+import clojure.lang.IFn;
+import org.luaj.vm2.lib.BaseLib;
+
+import java.io.InputStream;
+
+public class DefoldBaseLib extends BaseLib {
+    private final IFn resourceFinder;
+
+    public DefoldBaseLib(IFn resourceFinder) {
+        this.resourceFinder = resourceFinder;
+    }
+
+    @Override
+    public InputStream findResource(String filename) {
+        return (InputStream) resourceFinder.invoke(filename);
+    }
+}
```

Comment:
This used to be a `(proxy BaseLib [] ...)`, but I decided to make it a class so it's more efficient.